src/pages/storage/CustomVolumeSelectModal.tsx

```diff
@@ -52,57 +53,61 @@ const CustomVolumeSelectModal: FC<Props> = ({
     { "aria-label": "Actions", className: "actions" },
   ];
 
-  const rows = volumes.map((volume) => {
-    return {
-      columns: [
-        {
-          content: (
-            <div className="u-truncate iso-name" title={volume.name}>
-              {volume.name}
-            </div>
-          ),
-          role: "cell",
-          "aria-label": "Name",
-        },
-        {
-          content: volume.pool,
-          role: "cell",
-          "aria-label": "Storage pool",
-        },
-        {
-          content: contentTypeForDisplay(volume),
-          role: "cell",
-          "aria-label": "Content type",
-        },
-        {
-          content: volume.used_by?.length,
-          role: "cell",
-          "aria-label": "Used by",
-        },
-        {
-          content: (
-            <Button
-              onClick={() => handleSelect(volume)}
-              dense
-              appearance={
-                primaryVolume?.name === volume.name &&
-                primaryVolume.type === volume.type &&
-                primaryVolume.pool == volume.pool
-                  ? "positive"
-                  : ""
-              }
-              aria-label={`Select ${volume.name}`}
-            >
-              Select
-            </Button>
-          ),
-          role: "cell",
-          "aria-label": "Actions",
-          className: "u-align--right",
-        },
-      ],
-    };
-  });
+  const rows = isFetching
+    ? []
+    : volumes
```

Comment:
the sort and map potentially can get expensive depending on number of volumes in the project, maybe worth while applying useMemo for the sorting?

Reply:
I think this component should only rerender, when `volumes` or `isFetching` changes. And in both cases, we have to calculate the rows again anyhow. I did a quick test with the possible dependencies of useMemo by adding this to the component:
```
  const foo = useMemo(() => {
    console.log('memo refresh');
  }, [volumes, isFetching]);
  console.log("rerender");
```
There is just one render cycle where the memo would not refresh. Probably better to keep it simple?